dask/dataframe/groupby.py

```diff
@@ -1099,13 +1095,22 @@ def __init__(
             self.observed["observed"] = observed
 
         self._meta = self.obj._meta.groupby(
-            index_meta, group_keys=group_keys, **self.observed, **self.dropna
+            by_meta, group_keys=group_keys, **self.observed, **self.dropna
         )
 
+    @property
+    @_deprecated(use_instead="Groupby.by")
```

Comment:
A little odd that we exposed this in the first place; was this ever a part of the pandas API? I don't think pandas exposes `by` either. I'm not sure it should be a public attribute for us?

Reply:
Yeah I totally agree that it probably shouldn't be public. Pandas doesn't expose it. At the moment all the kwargs that you pass into a groupby become public properties of the class. We should probably change that, but I'm not sure if it belongs in this PR.